### DISCUSSION TITLE
Delegator rewards in cosmoshub-4?

### DISCUSSION THREAD

**Discussion by wojciechowskip** [graphprotocol/graph-tooling#1483] [General]
Hi guys! Are you aware if that's possible in **Cosmoshub-4** to calculate **delegator rewards every block**?

I know that I can listen to events of validator rewards and validator proposer rewards every block. That's a lot already but I would like to go step further and calculate delegator reward every blcck.

To accomplish that I need proportion of shares between delegator and validator: 
delegatorTokensEarnedPerBlock = delegatorShares / validatorShares * tokensEarnedPerBlock

Is is possible to somehow get this information about shares? Does maybe cosmos sdk emit event which is re-calculating current shares? I havent really found usefull info about in in docs.

Thanks in advance!

**Comment by matthewdarwin** [comment]
This issue doesn't seem related to graph-tooling.
Re-open if you disagree.
